proxy/coco/src/request/waiting_room.rs

```diff
@@ -401,7 +401,7 @@ impl<T, D> WaitingRoom<T, D> {
             .filter_by_state(RequestState::Requested)
             .find(move |(_, request)| {
                 request.elapsed(timestamp.clone()) >= self.config.delta.clone()
-                    || request.attempts().clones == Clones::new(0)
+                    && request.attempts().clones == Clones::new(0)
```

Comment:
The `||` was saying, "If this is in a fresh state for cloning -- i.e. we haven't made an attempt yet -- then we should return it".

Here's a case analysis:
* elapsed < delta || clones == 0 ~= false || true => returns the request
* elapsed > delta || clones == 0 ~= true || true => returns the request
* elapsed < delta || clone > 0 ~= false || false => does not return request
* elapsed > delta || clone > 0 ~= true || false => returns the request

Reply:
After reading this I realized I saw the diff the way around. The original issue states that "When searching for a project on the network the TimedOut state is hit too quickly.". How does this change address that, @xla? :thinking:

Reply:
> How does this change address that, @xla? thinking

By always returning the request even so its timeout hasn't elapsed yet. So it would be requiered immediately in short order until max queries are hit, meaning it would go into TimedOut as quick as we poll the waiting room. You can see that in the test case, which failed before this change, as it communicates the expectations and order of events clearly.

Reply:
I think the check for clones is completely wrong then and shouldn't even be in there. We don't want to prevent requests that fall back into `Requested` (by exhausting peer ids for clone attempts) to be tried again.

Could you add a test that does the following:
1. `let _ = waiting_room.request(&urn, 0);`
2. `queried`
3. `found`
4. `clone_failed`
5. `next_query` after delta amount of time

I expect that to fail with the check for clones in the code. If so, we should remove that line. If not, I'd like to try to understand why :smile:

Reply:
But ya, I was wrong about the `clones` check. Can't remember what I was optimising for there.

Reply:
I follow your logic and understand now how the oiginal check came to be, will add anothr test for the case when we wanna fall back to query after a clone failed and see if it holds.